cfme/infrastructure/virtual_machines.py

```diff
@@ -1339,9 +1353,21 @@ def genealogy(self):
 
 
 @attr.s
-class InfraTemplateCollection(TemplateCollection):
+class InfraTemplateCollection(ComparableMixin, TemplateCollection):
     ENTITY = InfraTemplate
 
+    COMPARE_VIEW_PROVIDER = TemplatesCompareView
+    COMPARE_VIEW_ALL = VmTemplatesCompareView
+    DROPDOWN_TEXT = 'Compare Selected Templates'
+    NAV_STRING = 'TemplatesOnly'
+
+    @property
+    def name(self):
```

Comment:
I would call this `provider_name`

Reply:
agreed. Updated.

Reply:
I could also add some conditionals in the code that references self.contect['object'] but that would seem to me to undo any clarity or readability we'd gain by renaming the property.

Reply:
I think I would update the property to provider_name and then add a variable name that would be `name = provider_name' .  That wouldn't clarify what .name is when calling it, but when looking at the class you would see that it's provider_name. Not sure how useful that would be.

Reply:
I could update the property to provider_name and then add a class variable name that would be `name = provider_name' . That wouldn't clarify what .name is when calling it, but when looking at the class you would see that it's provider_name. Not sure how useful that would be.

Reply:
Whatever needs this name (view, nav class) should either base the attribute lookup on the type, or should be using `self.parent` to get the name attribute.

```
name = getattr(self.context['object'], 'name', None) or getattr(self.context['object'].parent, 'name', None)
```

Reply:
OK. I updated the view that needs this for another reason and now I should include support for removing the name property. In progress.